### PR TITLE
ClearlyDefinedPackageCurationProvider: Minor code style fixes

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt [hunk 1]
```diff
@@ -196,15 +196,9 @@ private fun SourceLocation?.toArtifactOrVcs(): Any? =
...
 
             else -> {
-                val url = sourceLocation.url ?: run {
-                    when (sourceLocation.provider) {
-                        // TODO: Implement provider-specific mapping of coordinates to URLs.
-                        else -> ""
-                    }
-                }
-
+                // TODO: Implement provider-specific mapping of coordinates to URLs.
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt]: I believe the `TODO` should go above line 201 now, which will also make it look nicer in the following change.

File: analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt [hunk 2]
```diff
@@ -186,21 +186,16 @@ class ClearlyDefinedPackageCurationProvider(
...
-                    hash = Hash.NONE
-                )
-            }
+            ComponentType.GIT -> VcsInfoCurationData(
+                type = VcsType.GIT,
+                url = sourceLocation.url,
+                revision = sourceLocation.revision,
+                path = sourceLocation.path
+            )
+            // TODO: Implement provider-specific mapping of coordinates to URLs.
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt]: With the previous comment applied, the `TODO` should here land above line 197, and current line 195 should stay empty to separate cases.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — DISMISSED** [review]

**Comment by fviernau** [comment]
@sschuberth just updated the commit message titles for semantic commits.

File: analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt [hunk 3]
```diff
@@ -159,7 +159,7 @@ class ClearlyDefinedPackageCurationProvider(
                 runCatching { declaredLicense.toSpdx(SpdxExpression.Strictness.ALLOW_CURRENT) }.getOrNull()
             }
 
-            val sourceLocation = curation.described?.sourceLocation.toArtifactOrVcs()
+            val sourceLocation = curation.described?.sourceLocation?.toArtifactOrVcs()
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt:162]: Nit: This commit could even be of "refactor" type.

**Review by sschuberth — APPROVED** [review]

**Review by sschuberth — APPROVED** [review]

**Pull request merged**